src/firefly/js/visualize/projection/WavelengthHeaderParser.js

```diff
@@ -19,21 +21,147 @@ const allGoodValues= (...numbers)  => numbers.every((n) => !isNaN(n));
 
 const L_10= Math.log(10);
 
-function calculateWavelengthParams(parse, altWcs, wlTable) {
+/**
+ * According to A&A 395, 1061-1075 (2002) DOI: 10.1051/0004-6361:20021326
+ * Representations of world coordinates in FITS E. W. Greisen1 - M. R. Calabretta2
+ * paper,  the CD_ij is for the older FITs header. The newer FITs header has PC only.
+ * 1. If both PC and CD exist, it is wrong.  Exception is thrown
+ * 2. If PC is not exist, one or more CD is exist, use CD_ij (j=1..N); if any of CD_ij
+ *   is not defined, the default is 0.0;
+ * 3. If any PC_ij is defined or neigther PC nor CD is defined, we use PC, the default
+ *    is 0 for j!=i and 1 if j==i
+ *
+ * @param header
+ * @param zeroHeader
+ * @param which
+ * @returns {string}
+ */
+function getPC_ijKey(header, zeroHeader,which){
+
+    var key;
+    let hasPC=false, hasCD=false;
+    for (key in header) {
+        if (key.startsWith(`PC${which}_`)) {
+            hasPC = true;
+            break;
+        }
+    }
+    if (!hasPC && zeroHeader){
+        for (key in zeroHeader) {
+            if (key.startsWith(`PC${which}_`)) {
+                hasPC = true;
+                break;
+            }
+        }
+    }
+    for (key in header) {
+        if (key.startsWith(`CD${which}_`)) {
+            hasCD = true;
+            break;
+        }
+    }
+    if (!hasCD && zeroHeader){
+        for (key in zeroHeader) {
+            if (key.startsWith(`CD${which}_`)) {
+                hasCD = true;
+                break;
+            }
+        }
+    }
+    if (hasPC && hasCD){
+        throw Error('The FITs header contains both PC_ij and CD_ij');
```

Comment:
We don't want to throw exceptions here, That has the potential to break all the new image processing. Just return a a `false` or something and return an `undefined` in the calling function.